table/tables/index.go

```diff
@@ -314,10 +345,15 @@ func (c *index) Create(sctx sessionctx.Context, rm kv.RetrieverMutator, indexedV
 		if err != nil {
 			return nil, err
 		}
-		idxVal = make([]byte, 1+len(rowRestoredValue))
-		copy(idxVal[1:], rowRestoredValue)
+		// tailLen(1) + commonHandleFlag(1) + handleLen(2) + handle + restoredValue
+		idxValCap := 1 + 1 + 2 + h.Len() + len(rowRestoredValue)
+		idxVal = make([]byte, 1, idxValCap)
+		if !h.IsInt() && distinct {
+			idxVal = encodeCommonHandle(idxVal, h)
+		}
+		idxVal = append(idxVal, rowRestoredValue...)
 		tailLen := 0
-		if distinct {
+		if h.IsInt() && distinct {
 			// The len of the idxVal is always >= 10 since len (restoredValue) > 0.
 			tailLen += 8
 			idxVal = append(idxVal, EncodeHandleInUniqueIndexValue(h.IntValue())...)
```

Comment:
Should move this line before L354?

Reply:
No.
It's consistent with the comment.
Integer handle is appended at the tail if it has restored value.